From 0351cdfc9210f6f88863419534250ab7b8ce04ed Mon Sep 17 00:00:00 2001
From: ivan tkachenko <me@ratijas.tk>
Date: Mon, 25 Jul 2022 20:06:12 +0300
Subject: [PATCH] Fix mouse settings not being loaded when a mouse is connected

Keyboard daemon was calling into the wrong module name. There are only
so much kcminit modules, and all of them are required to be named and
referenced with a "kcm_" prefix.

Postmortem:

- Mouse settings' saving and loading worked fine.
- Changing settings on the fly worked fine, and if they were not loaded
  before due to this bug, KCM loaded them and applied, after which
  mouse worked fine until disconnected.
- Starting computer with mouse already connected worked fine.
- Plug&Play (i.e. connecting mouse after session has started) was the
  only thing that was broken, because this reconfiguration failed to
  launch the proper mouse module.

Co-Authored-By: David Edmundson <kde@davidedmundson.co.uk>
BUG: 435113
FIXED-IN: 5.24.7, 5.25.4, 5.26
(cherry picked from commit 0d7eccfbba7c3656487396dd381fcee0717bb1ec)
---
 kcms/keyboard/keyboard_daemon.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kcms/keyboard/keyboard_daemon.cpp b/kcms/keyboard/keyboard_daemon.cpp
index ae44b0ef2..56a93c4aa 100644
--- a/kcms/keyboard/keyboard_daemon.cpp
+++ b/kcms/keyboard/keyboard_daemon.cpp
@@ -89,7 +89,7 @@ void KeyboardDaemon::configureKeyboard()
 void KeyboardDaemon::configureMouse()
 {
     QStringList modules;
-    modules << QStringLiteral("mouse");
+    modules << QStringLiteral("kcm_mouse");
     QProcess::startDetached(QStringLiteral("kcminit"), modules);
 }
 
-- 
GitLab

